src/Native/Unix/configure.cmake

```diff
@@ -193,6 +193,11 @@ check_symbol_exists(
     "termios.h"
     HAVE_TCSANOW)
 
+check_symbol_exists(
+    SYS_getrandom
+    "sys/syscall.h;sys/random.h"
+    HAVE_SYS_GETRANDOM)
```

Comment:
Right now this is OK because we build the portable asset on a Linux distro that has this defined.  But if we were to change to build portable from CentOS7, RHEL6, or RHEL7 then this change would disappear.

I wonder if we should just locally replicate the syscall ID and definition of GRND_NONBLOCK to maintain flexibility on our build lab.  (With comments to explain why)

Reply:
I'm fine either way, but the existence of `sys/random.h` is tested here because of a comment @am11 made.  It was using syscall(2) directly before.

Reply:
Well, I think that the PR at the time had both the `check_symbol_exists` for `SYS_getrandom` and a manual define for GRND_NONBLOCK.  I like them being treated the same, but my comment is to question if there should not be a `check_symbol_exists` / `HAVE_SYS_GETRANDOM`, but rather just

```C
#if LINUX

#ifdef __NR_getrandom
static_assert(__NR_getrandom == 278);
#endif

// Equivalent to SYS_getrandom, but sidesteps the forward/re-definition since there are both
// pre-3.14 and post-3.14 kernels in the distro support matrix.
#define PAL_getrandom 278
```

Since the fallback for it not existing (if we changed our setup) is to just use urandom, perhaps this is good (and better than a forward define, since it avoids assuming the syscall ID on any future architectures) as-is.

Merely an exploratory question.

Reply:
Not really a good idea to hardcode syscall values: they're arch-specific.

Reply:
In any case, yes, doing some conditional platform-specific compilation is fine by me.  Kind of a po-tay-to/po-tah-to situation; I'm fine either way.

Reply:
Resolved.

Reply:
Can you please verify that the `getrandom` syscall path is going to be present in the binaries that we ship that are built on CentOS7?

> Not really a good idea to hardcode syscall numbers: they're arch-specific

Yes, but it won't be the first place we do this: https://github.com/dotnet/coreclr/blob/0463767cd7362a0044c4c7af0feda5bd9ad94804/src/gc/unix/gcenv.unix.cpp#L46 . I do not think there is a better way to do the light up for newer kernels on Linux.

cc @janvorli

Reply:
This has been resolved.  `syscall()` is used now instead of `getrandom()`.